### PR TITLE
[sqlsrv] database name with dot

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #27438
all credits to @dawe78
see #27462
### Summary of Changes
use `[]`

### Testing Instructions
Connect to a MSSQL database using sqlsrv-driver; database name contains dots: database.v.3.2.4

### Expected result
connect to database server, select database

### Actual result
Error message "Could not connect to SQL Server database."

### CODE REVIEW

File: libraries/joomla/database/driver/sqlsrv.php [hunk 1]
```diff
@@ -772,7 +772,7 @@ public function select($database)
 			return false;
 		}
 
-		if (!sqlsrv_query($this->connection, 'USE ' . $database, null, array('scrollable' => SQLSRV_CURSOR_STATIC)))
+		if (!sqlsrv_query($this->connection, 'USE [' . $database .']', null, array('scrollable' => SQLSRV_CURSOR_STATIC)))
```
Comment (Quy) [inline libraries/joomla/database/driver/sqlsrv.php]:
```suggestion
		if (!sqlsrv_query($this->connection, 'USE [' . $database . ']', null, array('scrollable' => SQLSRV_CURSOR_STATIC)))
```

**Comment by dawe78** [comment]
Thanks guys!

**Comment by HLeithner** [comment]
> Thanks guys!

Can you please test this?

**Comment by richard67** [comment]
@dawe78 Could you test this PR? I would test, but I can't because I don't have MS SQL Server available. We don't have many people using MS SQL Server and so have problems to find testers for it.

**Comment by dawe78** [comment]
I'm sorry, I'm very busy at the moment... I tested it yesterday and it works perfect!

**Comment by richard67** [comment]
@dawe78 Could you mark your test result in the issue tracker? Just go to [https://issues.joomla.org/tracker/joomla-cms/27484](https://issues.joomla.org/tracker/joomla-cms/27484) and use the "Test this" button at the top left corner. Then select the appropriate test result (success) and submitt. Thanks in advance.

**Comment by dawe78** [comment]
I have tested this item :white_check_mark: successfully on e6ef38017acf827e5b8a6febed7c937dc6e1f667<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27484">issues.joomla.org/tracker/joomla-cms/27484</a>.</sub>

**Comment by toivo** [comment]
Wasn't SQL Server support going to be dropped in Joomla 4?  

Ref. https://docs.joomla.org/Potential_backward_compatibility_issues_in_Joomla_4

**Comment by SharkyKZ** [comment]
@toivo That's correct. But this patch is for 3.x which still accepts bug fixes.

**Comment by richard67** [comment]
Question is if we should have the same change for other db types in J4.

I remember I have seen database names with dots in past in an industrial environment, too.

So it seems to be not so exotic.

**Comment by brianteeman** [comment]
mysql does not allow dots

**Comment by richard67** [comment]
> mysql does not allow dots

I see, that's right. So nothing to do for this in J4.

**Comment by HLeithner** [comment]
We should add it to the frame work https://github.com/joomla-framework/database/blob/2.0-dev/src/Sqlsrv/SqlsrvDriver.php#L137

**Comment by alikon** [comment]
done https://github.com/joomla-framework/database/pull/210

**Comment by HLeithner** [comment]
Thanks for the framework pr

**Comment by alikon** [comment]
merged in the framework not in the staging ?
let's close it

**Comment by HLeithner** [comment]
As always we need tests...

**Comment by richard67** [comment]
@HLeithner We have 1 good test by the one who made the issue. Will be hard to get a 2nd tester for MS SQL Server. I suggest to reopen and merge by review.

**Comment by alikon** [comment]
that's why i'm closing 
i wish you good luck on finding 2 mssql user's that test this

**Comment by richard67** [comment]
@alikon In opposite to J4, in J3 we don't fetch regularly the package from the framework, so it should be fixed here too.

**Comment by alikon** [comment]
sorry i've decided to clean my backlog 
it is not a template issues ... so who cares

**Comment by HLeithner** [comment]
So a second can't be too hard ;)

**Comment by richard67** [comment]
we have already 1 good test

**Comment by alikon** [comment]
anyway you can find some payed contributor to do this silly work
:angel:

**Pull request closed without merging**